### ISSUE TITLE
consider speechSynthesis API

### ISSUE THREAD

**Issue by pkra** [krautzource/aria-tree-walker#27]
It's almost well supported nowadays. Could we do something here or is this more a downstream issuee?

**Comment by pkra** [comment]
To be clear, there's a lot wrong with this. It's just a quick demo to see the basics.

**Comment by pkra** [comment]
Oh, and this was all inspired by https://brichwin.pages.iu.edu/canvasBookmarklets.html which also has this lovely hack for avoiding schwa for `a`:  `msg = msg.trim().replace(/\ba\b/g, 'A');`

**Comment by pkra** [comment]
Here's a different approach to work independently of navigator, using mutation observers.

```js
      const observer = new MutationObserver(function(mutationRecordArray) {
        const activeDescendantRecord = mutationRecordArray.find(record => record.target.getAttribute('tabindex') === '0');
        if (!activeDescendantRecord) {
          window.speechSynthesis.cancel();
          return;
        }
        const activeDescendant = activeDescendantRecord.target;
        if (!activeDescendant.getAttribute('aria-label')) return;
        if (window.speechSynthesis.speaking) {
          window.speechSynthesis.cancel();
        }
        let ssu = new SpeechSynthesisUtterance(activeDescendant.getAttribute('aria-label'));
        window.speechSynthesis.speak(ssu);
      });

      document.querySelectorAll('[data-treewalker]').forEach(node => observer.observe(node, {subtree: true, attributeFilter: ['tabindex']}));
```

This has one shortcoming (when moving out of a tree), but that could be solved with a little care (or, I think, #32).
